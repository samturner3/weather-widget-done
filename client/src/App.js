import React, { Component, Fragment } from 'react';
import Flipcard from '@kennethormandy/react-flipcard';

import './styles/App.css';
import '@kennethormandy/react-flipcard/dist/Flipcard.css'

import Forecast from './components/Forecast';
import CurrentWeather from './components/CurrentWeather';
import BackSide from './components/BackSide';

class App extends Component {
		state = {
			currentWeatherData: null,
			country: null,
			forecastData: null,
			fiveDayForecast: null,
			placeData: null,
			backgroundURL: '/images/SydneyBackground.jpg',
			isFlipped: true,
			selectedLocationString: 'Sydney',
			weatherLoading: true,
			placesCache: null
		};

		componentDidMount = () => {
			this.fetchCache();
		}

		showBack = () => {
			this.setState({
				isFlipped: true
			});
			this.fetchCache();
		};

		showFront = () => {
			this.setState({
				isFlipped: false
			});
		};

		fetchWeather = async (place, lat, lng) => {
			let backgroundURL = null;
			if (place.photos !== undefined && place.photos.length > 0) {
				backgroundURL = place.photos[0].getUrl({ 'maxWidth': 1000, 'maxHeight': 1000 });
			}
			let requestOptions = new Request(`/api/weather`, {
				method: 'POST',
				headers: new Headers({
					'Accept': 'application/json',
					'Content-Type': 'application/json'
				}),
				body: JSON.stringify({
					placeID: place.place_id,
					backgroundURL,
					lat,
					lng
				})
			});
			await fetch(requestOptions)
				.catch((response) => { alert(`There was a problem fetching data from the server. Please refresh/restart server and try again, or inspect the network tab. ${response}`) })
				.then((response) => {
					response.json().then(async (data) => {
						// console.log(data);
						// console.log('returned data current weather:', data.weather);
						this.setState({
							currentWeatherData: data.weather,
							forecastData: data.forecast
						});
						if (place.photos !== undefined && place.photos.length > 0) {
							this.setState({
								backgroundURL: `backgroundImages/${place.place_id}.jpg`
							});
						} else {
							this.setState({
								backgroundURL: null
							});
						}
						this.extractForecast(data.forecast);
					});
				});
		}

		setLocation = async (place, lat, long) => {
			this.setState({
				placeData: place,
				weatherLoading: true,
				isFlipped: false,
				selectedLocationString: place.address_components[0].long_name
			});
			await this.fetchWeather(place, lat, long);
			this.extractCountry(place);
			this.setState({
				weatherLoading: false
			});
		}

		fetchCache = () => {
			fetch('/api/cachelookup')
				.then(response => response.json())
				.then(data => {
					this.setState({ placesCache: data.cache });
				});
		};

		extractCountry = (placeData) => {
			placeData.address_components.forEach((addressComponent) => {
				addressComponent.types.forEach((type) => {
					if (type === 'country') {
						this.setState({
							country: addressComponent.long_name
						})
					}
				})
			})
		};

		extractForecast = (forecastData) => {
			const fiveDayForecast = [];
			forecastData.list.forEach((listItem) => {
				if (listItem.dt_txt.substr(listItem.dt_txt.length - 8) === '12:00:00') { // Grab the 12Noon forecast
					const dayForecast = {};
					dayForecast.time = listItem.dt_txt;
					dayForecast.description = listItem.weather[0].description;
					dayForecast.id = listItem.weather[0].id;
					dayForecast.temp = listItem.main.temp;
					dayForecast.temp_max = listItem.main.temp_max;
					dayForecast.temp_min = listItem.main.temp_min;
					fiveDayForecast.push(dayForecast);
				}
			})
			this.setState({
				fiveDayForecast: fiveDayForecast
			});
		};

		render() {
			return (
				<Fragment>
					<Flipcard flipped={this.state.isFlipped}>
						<div className='App'>
							<div className='container' ref='flipper'>
								<div className='widget_contain' style={{backgroundImage: `url(${this.state.backgroundURL})`}}>
									<div className='widget_left'>
										{this.state.placeData && (
											<CurrentWeather currentWeatherData={this.state.currentWeatherData} weatherLoading={this.state.weatherLoading} placeData={this.state.placeData} country={this.state.country} showBack={this.showBack} />
										)}
									</div>
									<div className='widget_right'>
										{this.state.fiveDayForecast && (
											<Forecast fiveDayForecast={this.state.fiveDayForecast} weatherLoading={this.state.weatherLoading} />
										)}
									</div>
								</div>
							</div>
						</div>
						<BackSide showFront={this.showFront} setLocation={this.setLocation} weatherData={this.state.weatherData} placesCache={this.state.placesCache} />
					</Flipcard>
				</Fragment>
			);
		}
}

export default App;
