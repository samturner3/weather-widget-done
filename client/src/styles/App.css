html, body {
  height: 100%; }

body {
  margin: 0; }

.App, #root, .Flipcard, .Flipcard-flipper, .Flipcard-front, .Flipcard-back {
  height: 100%;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%; }

.App-logo {
  -webkit-animation: App-logo-spin infinite 20s linear;
  animation: App-logo-spin infinite 20s linear;
  height: 80px; }

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white; }

.App-title {
  font-size: 1.5em; }

.App-intro {
  font-size: large; }

@-webkit-keyframes App-logo-spin {
  from {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg); }
  to {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg); } }

@keyframes App-logo-spin {
  from {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg); }
  to {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg); } }

.container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center; }

.current_weather_icon {
  font-size: 100px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column; }
  @media only screen and (max-width: 600px) {
    .current_weather_icon {
      font-size: 50px; } }
  .current_weather_icon-top {
    -webkit-box-flex: 1;
    -ms-flex: 1;
    flex: 1; }
  .current_weather_icon-bottom {
    -webkit-box-flex: 1;
    -ms-flex: 1;
    flex: 1; }
  .current_weather_icon i {
    padding-right: 20px; }

.widget_contain {
  color: white;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap; }

.widget_left {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 7;
  -ms-flex-positive: 7;
  flex-grow: 7;
  background: rgba(0, 0, 0, 0.5);
  padding: 15px; }

.widget_left_item {
  min-width: 200px;
  height: 30px;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-box-flex: 1;
  -ms-flex: 1;
  flex: 1; }
  .widget_left_item-top {
    -webkit-box-flex: 1;
    -ms-flex: 1;
    flex: 1; }
    .widget_left_item-top p {
      margin: 8px 0 8px 0; }
  .widget_left_item-bottom {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    flex-direction: column;
    margin-bottom: 10px; }

.widget_right {
  background: rgba(0, 0, 0, 0.75);
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 5;
  -ms-flex-positive: 5;
  flex-grow: 5;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 40px; }
  @media only screen and (max-width: 600px) {
    .widget_right {
      padding: 15px; } }
  .widget_right_item {
    width: 100%;
    min-width: 100px;
    height: 30px;
    -ms-flex-item-align: center;
    align-self: center;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-align: end;
    -ms-flex-align: end;
    align-items: flex-end;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
    border-bottom: 1px solid white;
    -webkit-box-flex: 5;
    -ms-flex-positive: 5;
    flex-grow: 5;
    padding-bottom: 10px;
    padding-top: 10px;
    margin-top: 10px; }
    .widget_right_item_topRow {
      -webkit-box-flex: 2;
      -ms-flex: 2;
      flex: 2;
      display: -webkit-box;
      display: -ms-flexbox;
      display: flex;
      -webkit-box-orient: vertical;
      -webkit-box-direction: normal;
      -ms-flex-direction: column;
      flex-direction: column; }
    .widget_right_item-description {
      -webkit-box-flex: 1;
      -ms-flex: 1;
      flex: 1; }
    .widget_right_item-temp {
      -webkit-box-flex: 1;
      -ms-flex: 1;
      flex: 1; }
    .widget_right_item-day {
      margin-left: auto; }
    .widget_right_item_bottomRow {
      -webkit-box-flex: 1;
      -ms-flex: 1;
      flex: 1;
      display: -webkit-box;
      display: -ms-flexbox;
      display: flex; }
