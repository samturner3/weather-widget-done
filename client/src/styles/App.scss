html, body {
  height: 100%;
}
body {
  margin: 0;
}

.App, #root, .Flipcard, .Flipcard-flipper, .Flipcard-front, .Flipcard-back {
  height: 100%;
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  width: 100%;
}

.App-logo {
  -webkit-animation: App-logo-spin infinite 20s linear;
          animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@-webkit-keyframes App-logo-spin {
  from { -webkit-transform: rotate(0deg); transform: rotate(0deg); }
  to { -webkit-transform: rotate(360deg); transform: rotate(360deg); }
}

@keyframes App-logo-spin {
  from { -webkit-transform: rotate(0deg); transform: rotate(0deg); }
  to { -webkit-transform: rotate(360deg); transform: rotate(360deg); }
}

.container {
  // width: 600px;
  // height: 420px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}

.current_weather_icon {
  @media only screen and (max-width: 600px) {
    font-size: 50px;
  }
  font-size: 100px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  &-top {
    -webkit-box-flex: 1;
        -ms-flex: 1;
            flex: 1;
  }
  &-bottom {
    -webkit-box-flex: 1;
        -ms-flex: 1;
            flex: 1;
  }
  i {
    padding-right: 20px;
  }
}

.widget_contain {
  color: white;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -ms-flex-wrap: wrap;
      flex-wrap: wrap;
}

.widget_left {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex; 
  -webkit-box-orient: vertical; 
  -webkit-box-direction: normal; 
      -ms-flex-direction: column; 
          flex-direction: column;
  -webkit-box-flex: 7;
      -ms-flex-positive: 7;
          flex-grow: 7;
  background: rgba(0, 0, 0, 0.5);
  // width: 400px;
  padding: 15px;
}

.widget_left_item {
  min-width: 200px;
  height: 30px;
  -ms-flex-item-align: center;
      align-self: center;
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
    &-top {
      -webkit-box-flex: 1;
          -ms-flex: 1;
              flex: 1;
      p {
        margin: 8px 0 8px 0;
      }
    }
    &-bottom {
      display: -webkit-box;
      display: -ms-flexbox;
      display: flex;
      -webkit-box-orient: vertical;
      -webkit-box-direction: normal;
          -ms-flex-direction: column;
              flex-direction: column;
      margin-bottom: 10px;
    }
  }

.widget_right {
  background: rgba(0, 0, 0, .75);
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-flex: 5;
      -ms-flex-positive: 5;
          flex-grow: 5;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  @media only screen and (max-width: 600px) {
    padding: 15px;
  }
  padding: 40px;
    &_item {
      width: 100%;
      min-width: 100px;
      height: 30px;
      -ms-flex-item-align: center;
          align-self: center;
      display: -webkit-box;
      display: -ms-flexbox;
      display: flex;
      -webkit-box-align: end;
          -ms-flex-align: end;
              align-items: flex-end;
      -webkit-box-pack: center;
          -ms-flex-pack: center;
              justify-content: center;
      border-bottom: 1px solid white;
      -webkit-box-flex: 5;
          -ms-flex-positive: 5;
              flex-grow: 5;
      padding-bottom: 10px;
      padding-top: 10px;
      margin-top: 10px;
      &_topRow {
        -webkit-box-flex: 2;
            -ms-flex: 2;
                flex: 2;
        display: -webkit-box;
        display: -ms-flexbox;
        display: flex;
        -webkit-box-orient: vertical;
        -webkit-box-direction: normal;
            -ms-flex-direction: column;
                flex-direction: column;
      }
      &-description {
        -webkit-box-flex: 1;
            -ms-flex: 1;
                flex: 1;
      }
      &-temp {
        -webkit-box-flex: 1;
            -ms-flex: 1;
                flex: 1;
      }
      &-day {
        margin-left: auto;
      }
      &_bottomRow {
        -webkit-box-flex: 1;
            -ms-flex: 1;
                flex: 1;
        display: -webkit-box;
        display: -ms-flexbox;
        display: flex;
      }
    }
}