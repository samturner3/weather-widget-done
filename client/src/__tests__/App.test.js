import React from 'react';
import App from '../App';

import {
	selectedPlaceMock,
	forecastDataMock,
	fiveDayForecastMock
} from './__mocks__/mockData'

function setup() {
	return {
		wrapper: shallow(<App />)
	};
}

describe('<Clock />', () => {
	beforeAll(() => {
		window.fetch = jest.fn().mockImplementation(() =>
			Promise.resolve({
				ok: true,
				json: () => [{
					'key': '-33.8688197+151.20929550000005',
					'name': 'Sydney',
					'exp': 1535207229409
				}]
			})
		);
	});

	it('should render all elements', () => {
		const { wrapper } = setup();
		expect(wrapper).toMatchSnapshot();
	});

	describe('showBack()', () => {
		it('should set state and call the fetch catche method', () => {
			const { wrapper } = setup();

			const { isFlipped } = wrapper.instance().state;
			const fetchCacheSpy = jest.spyOn(wrapper.instance(), 'fetchCache');

			wrapper.instance().showBack();

			expect(isFlipped).toBe(true);
			expect(fetchCacheSpy).toHaveBeenCalledTimes(1);
		});
	});

	describe('showFront()', () => {
		it('should set state.isFlipped to false', () => {
			const { wrapper } = setup();
			wrapper.instance().showFront();

			const { isFlipped } = wrapper.instance().state;

			expect(isFlipped).toBe(false);
		});
	});

	describe('extractCountry()', () => {
		it('should set state.country', () => {
			const { wrapper } = setup();
			wrapper.instance().extractCountry(selectedPlaceMock);

			const { country } = wrapper.instance().state;

			expect(country).toBe('Australia');
		});
	});

	describe('extractForecast()', () => {
		it('should set state.fiveDayForecast', () => {
			const { wrapper } = setup();
			wrapper.instance().extractForecast(forecastDataMock);

			const { fiveDayForecast } = wrapper.instance().state;

			expect(fiveDayForecast).toHaveLength(fiveDayForecastMock.length);

			expect(JSON.stringify(fiveDayForecast)).toBe(JSON.stringify(fiveDayForecastMock));
		});
	});
});
