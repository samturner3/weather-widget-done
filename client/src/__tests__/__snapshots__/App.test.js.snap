// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> should render all elements 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Flipcard
        classNameBack=""
        classNameFlipper=""
        classNameFront=""
        disabled={false}
        flipped={true}
        type="horizontal"
>
        <div
                className="App"
        >
                <div
                        className="container"
                >
                        <div
                                className="widget_contain"
                                style={
                                        Object {
                                                "backgroundImage": "url(/images/SydneyBackground.jpg)",
                                              }
                                }
                        >
                                <div
                                        className="widget_left"
                                />
                                <div
                                        className="widget_right"
                                />
                        </div>
                </div>
        </div>
        <BackSide
                placesCache={null}
                setLocation={[Function]}
                showFront={[Function]}
                weatherData={undefined}
        />
</Flipcard>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="App"
>
            <div
                        className="container"
            >
                        <div
                                    className="widget_contain"
                                    style={
                                                Object {
                                                            "backgroundImage": "url(/images/SydneyBackground.jpg)",
                                                          }
                                    }
                        >
                                    <div
                                                className="widget_left"
                                    />
                                    <div
                                                className="widget_right"
                                    />
                        </div>
            </div>
</div>,
          <BackSide
            placesCache={null}
            setLocation={[Function]}
            showFront={[Function]}
            weatherData={undefined}
/>,
        ],
        "classNameBack": "",
        "classNameFlipper": "",
        "classNameFront": "",
        "disabled": false,
        "flipped": true,
        "type": "horizontal",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
>
              <div
                            className="widget_contain"
                            style={
                                          Object {
                                                        "backgroundImage": "url(/images/SydneyBackground.jpg)",
                                                      }
                            }
              >
                            <div
                                          className="widget_left"
                            />
                            <div
                                          className="widget_right"
                            />
              </div>
</div>,
            "className": "App",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="widget_contain"
                style={
                                Object {
                                                "backgroundImage": "url(/images/SydneyBackground.jpg)",
                                              }
                }
>
                <div
                                className="widget_left"
                />
                <div
                                className="widget_right"
                />
</div>,
              "className": "container",
            },
            "ref": "flipper",
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="widget_left"
/>,
                  <div
                    className="widget_right"
/>,
                ],
                "className": "widget_contain",
                "style": Object {
                  "backgroundImage": "url(/images/SydneyBackground.jpg)",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": null,
                    "className": "widget_left",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": null,
                    "className": "widget_right",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "placesCache": null,
            "setLocation": [Function],
            "showFront": [Function],
            "weatherData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Flipcard
          classNameBack=""
          classNameFlipper=""
          classNameFront=""
          disabled={false}
          flipped={true}
          type="horizontal"
>
          <div
                    className="App"
          >
                    <div
                              className="container"
                    >
                              <div
                                        className="widget_contain"
                                        style={
                                                  Object {
                                                            "backgroundImage": "url(/images/SydneyBackground.jpg)",
                                                          }
                                        }
                              >
                                        <div
                                                  className="widget_left"
                                        />
                                        <div
                                                  className="widget_right"
                                        />
                              </div>
                    </div>
          </div>
          <BackSide
                    placesCache={null}
                    setLocation={[Function]}
                    showFront={[Function]}
                    weatherData={undefined}
          />
</Flipcard>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="App"
>
              <div
                            className="container"
              >
                            <div
                                          className="widget_contain"
                                          style={
                                                        Object {
                                                                      "backgroundImage": "url(/images/SydneyBackground.jpg)",
                                                                    }
                                          }
                            >
                                          <div
                                                        className="widget_left"
                                          />
                                          <div
                                                        className="widget_right"
                                          />
                            </div>
              </div>
</div>,
            <BackSide
              placesCache={null}
              setLocation={[Function]}
              showFront={[Function]}
              weatherData={undefined}
/>,
          ],
          "classNameBack": "",
          "classNameFlipper": "",
          "classNameFront": "",
          "disabled": false,
          "flipped": true,
          "type": "horizontal",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="container"
>
                <div
                                className="widget_contain"
                                style={
                                                Object {
                                                                "backgroundImage": "url(/images/SydneyBackground.jpg)",
                                                              }
                                }
                >
                                <div
                                                className="widget_left"
                                />
                                <div
                                                className="widget_right"
                                />
                </div>
</div>,
              "className": "App",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="widget_contain"
                  style={
                                    Object {
                                                      "backgroundImage": "url(/images/SydneyBackground.jpg)",
                                                    }
                  }
>
                  <div
                                    className="widget_left"
                  />
                  <div
                                    className="widget_right"
                  />
</div>,
                "className": "container",
              },
              "ref": "flipper",
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="widget_left"
/>,
                    <div
                      className="widget_right"
/>,
                  ],
                  "className": "widget_contain",
                  "style": Object {
                    "backgroundImage": "url(/images/SydneyBackground.jpg)",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": null,
                      "className": "widget_left",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": null,
                      "className": "widget_right",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "placesCache": null,
              "setLocation": [Function],
              "showFront": [Function],
              "weatherData": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
